esmvalcore/preprocessor/_io.py

```diff
@@ -174,6 +174,9 @@ def _get_concatenation_error(cubes):
 
 def concatenate(cubes):
     """Concatenate all cubes after fixing metadata."""
+    if len(cubes) == 1:
+        return cubes[0]
+
```

Comment:
are we calling `concatenate()` on **all** `cubes` irrespective of its length? In that case my bad I completely overseen the case of no need for concatenation :beer:

Reply:
Yes, it's one of the default preprocessor functions.